internal_packages/keybase/lib/pgp-key-store.cjsx

```diff
@@ -0,0 +1,391 @@
+NylasStore = require 'nylas-store'
+{DraftStore, MessageBodyProcessor, RegExpUtils} = require 'nylas-exports'
+{remote} = require 'electron'
+kb = require './keybase'
+pgp = require 'kbpgp'
+_ = require 'underscore'
+path = require 'path'
+fs = require 'fs'
+
+class PGPKeyStore extends NylasStore
+
+  constructor: ->
+    super()
+
+    @_pubKeys = {}
+    @_privKeys = []
+
+    @_msgCache = []
+    @_msgStatus = []
+
+    @_pubWatcher = null
+    @_privWatcher = null
+
+    @_keyDir = path.join(NylasEnv.getConfigDirPath(), 'keys')
+    @_pubKeyDir = path.join(@_keyDir, 'public')
+    @_privKeyDir = path.join(@_keyDir, 'private')
+
+    # Create the key storage file system if it doesn't already exist
+    fs.access(@_keyDir, fs.R_OK | fs.W_OK, (err) =>
```

Comment:
This looks great—I appreciate that it's all async (so it won't slow down the window loading.)

It might be interesting to consider the failure cases. For example, access + mkdir is nice and safe, but we could also try to mkdir all the time, and ignore the error if it already exists (since mkdir is non-destructive!)